src/main/scala/io/iohk/ethereum/network/PeerActor.scala

```diff
@@ -295,29 +297,51 @@ class PeerActor(
 
   class HandshakedHandler(rlpxConnection: RLPxConnection, initialStatus: msg.Status) {
 
+    var currentPeerMaxBlock: BigInt = 0
+
     def receive: Receive =
       handleSubscriptions orElse handleTerminated(rlpxConnection) orElse
       handlePeerChainCheck(rlpxConnection) orElse handlePingMsg(rlpxConnection) orElse
       handleBlockFastDownload(rlpxConnection) orElse {
       case RLPxConnectionHandler.MessageReceived(message) =>
         log.debug("Received message: {}", message)
+        updateMaxBlock(message)
         notifySubscribers(message)
         processMessage(message)
 
       case s: SendMessage[_] =>
+        updateMaxBlock(s.message)
         rlpxConnection.sendMessage(s.message)(s.enc)
 
+      case Broadcast(newBlock: NewBlock) =>
+        if (newBlock.block.header.number > currentPeerMaxBlock) {
+          rlpxConnection.sendMessage(newBlock)
+        }
+
       case GetStatus =>
         sender() ! StatusResponse(Handshaked(initialStatus))
     }
 
-    def notifySubscribers(message: Message): Unit = {
+    private def updateMaxBlock(message: Message) = message match {
+      case m: BlockHeaders =>
+        val maxBlockNumber = m.headers.map(_.number).fold(0: BigInt) { case (a, b) => if (a > b) a else b }
+        if (maxBlockNumber > currentPeerMaxBlock) {
+          currentPeerMaxBlock = maxBlockNumber
+        }
+      case m: NewBlock =>
+        if (m.block.header.number > currentPeerMaxBlock)
+          currentPeerMaxBlock = m.block.header.number
+          supervisor ! Broadcast(m)
+      case _ =>
```

Comment:
when we receive new block hash then we will ask for the header by this hash so we will update max block number